doc/src/explanation/gotchas.rst

```diff
@@ -806,4 +806,4 @@ These will give you the function parameters and docstring for
 
 .. module:: sympy.simplify.simplify
 .. autofunction:: powsimp
-   :noindex:
+   :no-index:
```

Comment:
This won't work for Sphinx 7.1 and earlier, I would reccommend staying with ``noindex`` until your lower bound on Sphinx is 7.2.

Reply:
I think it is fine. According to https://github.com/sphinx-doc/sphinx/pull/11533 `:noindex:` will be deprecated and removed at some point and I would rather just solve that problem now if I know about it.

Building docs is more of a development activity so we don't really need to support older versions of anything. Anyone who wants to do this can create a venv and use the requirements.txt. Ideally though we keep the versions unpinned and just make sure that the latest versions always work (CI always runs latest versions - the pin is only temporary).

Reply:
A user, like me, may want to use an older sphinx. It is nice if it can run with past versions to be more flexible.

Reply:
Okay, I've restored :noindex:

Now this PR just unpins Sphinx so let's see if it passes.

Reply:
Note that it takes `~1hr` to be able to see this.

Reply:
@moorepants I'm not clear why you would want to use an older Sphinx. I can see if you might need an older Sphinx for some other projects, or if you have some unrelated dependencies that pin sphinx, but if that's the case you should just use a separate environment for the SymPy docs. Some of our docs dependencies might pin Sphinx (I know Furo tends to), but other than that, we only need to pin it for our own environment if if causes some issue that we haven't been able to fix.

Reply:
When you are developing multiple packages (e.g. sympy + others) simultaneously it is preferable to use a single environment for that. When these packages depend on the same packages, then your environment can become unsolvable if some of the packages are too strict with dependency pins. Thus it is generally helpful to pin any dependencies (development or not) in the least restrictive possible way. Pinning to the absolute latest release of any dependency often causes issues for creating environments.